docs/running_lbann.rst

```diff
@@ -1,96 +1,434 @@
+
+  - During automatic differentiation ("backprop"), a layer receives
+    "input error signals" (objective function gradients w.r.t. output
+    tensors) from its children and sends "output error signals"
+    (objective function gradients w.r.t. input tensors) to its
+    parents. If the layer has any associated weights, it will also
+    compute objective function gradients w.r.t. the weights.
+
+  - Most layers require a specific number of parents and children, but
+    LBANN will insert layers into the graph if there is a mismatch and
+    the intention is obvious. For example, if a layer expects one
+    child but has multiple, then a split layer (with multiple output
+    tensors all identical to the input tensor) is inserted. Similarly,
+    if a layer has fewer children than expected, dummy layers will be
+    inserted. However, this does not work if there is any
+    ambiguity. In such cases (common with input and slice layers), it
+    is recommended to manually insert identity layers so that the
+    parent/child relationships are absolutely unambiguous.
+
++ Weights: A tensor consisting of trainable parameters, typically
+  associated with one or more layers. A weights owns an initializer to
+  initially populate its values and an optimizer to find values that
+  minimize the objective function.
+
+  - A weights without a specified initializer will use a zero
+    initializer.
+
+  - A weights without a specified optimizer will use the model's
+    default optimizer.
+
+  - If a layer requires weightses and none are specified, it will
+    create the needed weightses. The layer will pick sensible
+    initializers and optimizers for the weightses.
+
+  - The dimensions of a weights is determined by their associated
+    layers. The user can not set it directly.
+
+.. note:: It is unfortunate that the deep learning community has
+   settled upon the plural word "weights" to describe tensors of
+   trainable parameters. Rather than using awkward and ambiguous
+   phrases like "set of weights," we have given up on grammar and
+   refer to "weights" (singular) and "weightses" (plural).
+
++ Objective function: Mathematical expression that the optimizers will
+  attempt to minimize. It is made up of multiple terms that are added
+  together (possibly with scaling factors).
+
+  - An objective function term can get its value from a scalar-valued
+    layer, i.e. a layer with an output tensor with one entry.
+
++ Metric: Mathematical expression that will be reported to the
+  user. This typically does not affect training, but is helpful for
+  evaluating the progress of training.
+
++ Callback: Function that is performed at various points during an
+  experiment. Callbacks are helpful for reporting, debugging, and
+  performing advanced training techniques.
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Data readers
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. note:: The core infrastructure for data readers is slated for
```

Comment:
Again, for big changes we know are coming, `..warning::` might be better than `.. note::`